scripts/auth_certificates

```diff
@@ -147,15 +143,15 @@ if [[ "$AUTH_TYPE" = "ca" ]]; then
   # Add more info /C=$country/ST=$state/L=$locality/O=$organization/OU=$organizational_unit/CN=$common_name/emailAddress=$email
   SUBJ="/C=${SSL_COUNTRY}/ST=${SSL_STATE}/O=${SSL_ORGANIZATION}/CN=${AUTH_NAME}/emailAddress=${SSL_EMAIL}"
 
-  openssl genrsa -out "${SSL_LOCATION}/${AUTH_NAME}.key" 4096 2> /dev/null
+  openssl genrsa -out "${SSL_LOCATION}/${AUTH_NAME}.key" "${KEY_SIZE}" 2> /dev/null
   openssl req -new -x509 -days "${SSL_EXPIRATION_DAYS}" -key "${SSL_LOCATION}/${AUTH_NAME}.key" -out "${SSL_LOCATION}/${AUTH_NAME}.cert" -subj '/C=US/CN=MY-CA' 2> /dev/null
 
   certificate_log "info" "${AUTH_TYPE} certificate created successfully, and placed in ${SSL_LOCATION}" "outonly" "pretty"
 else
   # Add more info /C=$country/ST=$state/L=$locality/O=$organization/OU=$organizational_unit/CN=$common_name/emailAddress=$email
   SUBJ="/C=${SSL_COUNTRY}/ST=${SSL_STATE}/O=${SSL_ORGANIZATION}/CN=${SSL_NAME}/emailAddress=${SSL_EMAIL}"
 
-  openssl genrsa -out "${SSL_LOCATION}/${SSL_NAME}.key" 1024 2> /dev/null
+  openssl genrsa -out "${SSL_LOCATION}/${SSL_NAME}.key" "${KEY_SIZE}" 2> /dev/null
   openssl req -new -key "${SSL_LOCATION}/${SSL_NAME}.key" -subj "${SUBJ}" -out "${SSL_LOCATION}/${SSL_NAME}.csr" 2> /dev/null
   openssl x509 -req -days "${SSL_EXPIRATION_DAYS}" -in "${SSL_LOCATION}/${SSL_NAME}.csr" -CA "${SSL_LOCATION}/${AUTH_NAME}.cert" -CAkey "${SSL_LOCATION}/${AUTH_NAME}.key" -set_serial 01 -out "${SSL_LOCATION}/${SSL_NAME}.cert" 2> /dev/null
```

Comment:
Could you please check around `-CAcreateserial` vs `-set_serial 01`

Reply:
As per search I find below for their usage:

`-set_serial`
You want full control over the serial number (e.g., scripting, testing)	
You're scripting cert generation and want to avoid .srl file conflicts

`-CAcreateserial (it keeps track for you)`
You’re creating many certs over time with the same CA	

It appears that _set_serial_ is favorable among these.